sky/task.py

```diff
@@ -548,15 +548,30 @@ def from_yaml_config(
                              estimated_size_gigabytes=estimated_size_gigabytes)
 
         # Experimental configs.
-        experimnetal_configs = config.pop('experimental', None)
-        cluster_config_override = None
-        if experimnetal_configs is not None:
-            cluster_config_override = experimnetal_configs.pop(
+        experimental_configs = config.pop('experimental', None)
+
+        # Handle the top-level config field
+        config_override = config.pop('config', None)
+
+        # Handle backward compatibility with experimental.config_overrides
+        if experimental_configs is not None:
+            exp_config_override = experimental_configs.pop(
                 'config_overrides', None)
+            if exp_config_override is not None:
+                if config_override is not None:
+                    logger.warning(
+                        f'{colorama.Fore.YELLOW}Both top-level `config` and '
+                        f'`experimental.config_overrides` are specified. '
+                        f'Using top-level `config`.{colorama.Style.RESET_ALL}')
+                else:
+                    config_override = exp_config_override
```

Comment:
We should have a depreaction warning here with logger, when `exp_config_override` is used.